### DISCUSSION TITLE
Using Move hub with LWP3Device?

### DISCUSSION THREAD

**Discussion by doowttam** [pybricks/support#1877] [Q&A] [answered]
I'm trying to run a program on the Move hub to do hub-to-hub communication which is listed in the guide as supported: https://pybricks.com/learn/getting-started/what-do-you-need/#choosing-a-lego-hub

But, if I import the class I get an immediate import error. 

```python
from pybricks.hubs import MoveHub
from pybricks.pupdevices import Motor, ColorDistanceSensor
from pybricks.parameters import Button, Color, Direction, Port, Side, Stop
from pybricks.robotics import DriveBase
from pybricks.tools import wait, StopWatch

print('Start')

hub = MoveHub()
hub.light.on(Color.YELLOW)
wait(3000)

from pybricks.iodevices import LWP3Device

hub.light.on(Color.GREEN)
wait(3000)

print('Done')
```

Outputs:

```
Start
Traceback (most recent call last):
  File "train_control.py", in <module>
ImportError: 
```

The program fails at the same spot (stops running after yellow light) if I disconnect it from the computer and run it by pushing the button on the hub.

What am I doing wrong? Thanks in advance.

**Comment by dlech** [comment, marked as answer]
The `pybricks.iodevices` is not enabled on the BOOST Move hub due to limited flash memory on that hub.

The hub to hub communication mentioned in the docs is https://docs.pybricks.com/en/stable/hubs/movehub.html#bluetooth-examples, which is different from `LWP3Device`.

**Reply by doowttam** [reply]
I see. Thanks!

**Reply by vkreuz** [reply]
I encounter same problem, when trying to use Move hub as a controller for duplo train.
Is it possible to somehow enable LWP3Device on move hub in exchange of some other capability lost?
This should be probably mentioned in the docs: https://docs.pybricks.com/en/v3.5.0/iodevices/lwp3device.html

**Reply by dlech** [reply]
> Is it possible to somehow enable LWP3Device on move hub in exchange of some other capability lost?

No, I don't think there is anything else of equal size that we could trade for that is not an essential feature, like motors and sensors.



> This should be probably mentioned in the docs

It kind of is. It is on the `iodevices` module page - the entire module is not available. But since each class has it's own page, we should probably repeat that info on each page.

![image](https://github.com/user-attachments/assets/bd19b926-143b-4868-b317-5e634571912b)

**Reply by dlech** [reply]
> No, I don't think there is anything else of equal size that we could trade for that is not an essential feature, like motors and sensors

Well, you could compile your own firmware and do this. :smile: But not something we would do in a regular release.
